api/query/search.go

```diff
@@ -99,6 +99,25 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 			val, _ := shared.ParseBooleanParam(q, param)
 			isSimpleQ = isSimpleQ && (val == nil || !*val)
 		}
+		r2 := *r
+		r2url := *r.URL
+		r2.URL = &r2url
+		r2.Method = "GET"
+		runIDStrs := make([]string, 0, len(rq.RunIDs))
+		for _, id := range rq.RunIDs {
+			runIDStrs = append(runIDStrs, strconv.FormatInt(id, 10))
+		}
+		runIDsStr := strings.Join(runIDStrs, ",")
+		r2.URL.RawQuery = fmt.Sprintf("run_ids=%s&q=%s", url.QueryEscape(runIDsStr), url.QueryEscape(simpleQ.Pattern))
```

Comment:
r2.URL has a function called `[Query()](https://pkg.go.dev/net/url#URL.Query)`.
You should use that to get that existing query params.

Once you have that, you can use the [Add function](https://pkg.go.dev/net/url#Values.Add). And then re-encode it
```go
// do your loop
q.Add("run_ids",strconv.FormatInt(id, 10))

// any other query params

// finally add it back in
r2.URL.RawQuery = q.Encode()

```

Reply:
The issue here is that these run IDs live in `rq`, which is a `RunQuery` struct, and the run IDs aren't necessarily in the original query.
https://github.com/web-platform-tests/wpt.fyi/blob/8bf23a6f68d18acab002aa6a613fc5660afb0a85/api/query/atoms.go#L56-L62

That being said, I copied most of this code over from the another section of this function and I'm likely doing plenty wrong 😅 
https://github.com/web-platform-tests/wpt.fyi/blob/8bf23a6f68d18acab002aa6a613fc5660afb0a85/api/query/search.go#L149-L163

Reply:
i was thinking something like this:

```go
// Clone the existing request
r2 := r.Clone(r.Context())

// Extract the existing query parameters
q := r.URL.Query()

// Add specific query parameters that are missing [1/2]
for _, id := range rq.RunIDs {
  q.Add("run_ids", strconv.FormatInt(id, 10))
}

// Add specific query parameters that are missing [2/2]
q.Add("q", url.QueryEscape(simpleQ.Pattern))

// Encode into back into RawQuery
r2.URL.RawQuery = q.Encode()
```